apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/domain/Address.java

```diff
@@ -121,12 +154,14 @@ public boolean equals(Object obj) {
       if (obj == null || getClass() != obj.getClass()) return false;
       Address that = Address.class.cast(obj);
       return Objects.equal(this.addr, that.addr)
-               && Objects.equal(this.version, that.version);
+               && Objects.equal(this.version, that.version)
+               && Objects.equal(this.macAddr, that.macAddr)
+               && Objects.equal(this.type, that.type);
    }
    
    protected ToStringHelper string() {
       return Objects.toStringHelper(this)
-            .add("addr", addr).add("version", version);
+            .add("addr", addr).add("version", version).add("macAddr", macAddr).add("type", type);
```

Comment:
Use `macAddr.or("")` and `type.or("")` (or another default), to have a better string representation?